llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -3509,8 +3509,10 @@ static SDValue matchSplatAsGather(SDValue SplatVal, MVT VT, const SDLoc &DL,
     return SDValue();
 
   // Check that Index lies within VT
-  // TODO: Can we check if the Index is constant and known in-bounds?
-  if (!TypeSize::isKnownLE(Vec.getValueSizeInBits(), VT.getSizeInBits()))
+  if (auto *CIdx = dyn_cast<ConstantSDNode>(Idx)) {
+    if (VT.getVectorElementCount().getKnownMinValue() <= CIdx->getZExtValue())
+      return SDValue();
```

Comment:
Nit, it would read easier to me as "bail when the index is out of bounds", i.e.:
```suggestion
    if (CIdx->getZExtValue() >= VT.getVectorElementCount().getKnownMinValue())
      return SDValue();
```